llvm/lib/CheerpUtils/GlobalDepsAnalyzer.cpp

```diff
@@ -124,12 +125,35 @@ void GlobalDepsAnalyzer::extendLifetime(Function* F)
 	assert( visited.empty() );
 }
 
+static void replaceFunctionAliasWithAliasee(llvm::Module &module, StringRef name) {
+	GlobalAlias *funcAlias = module.getNamedAlias(name);
+	if (funcAlias) {
+		Function *actualFunc = dyn_cast<Function>(funcAlias->getAliaseeObject());
+
+		assert(actualFunc);
+
+		funcAlias->replaceAllUsesWith(actualFunc);
+		funcAlias->eraseFromParent();
+
+		ValueToValueMapTy VMap;
+		Function *newFunc = CloneFunction(actualFunc, VMap);
```

Comment:
Is the cloning done so that we don't have to worry about multiple aliases for the same function when changing the name?